src/nbl/video/CVulkanConnection.h

```diff
@@ -16,22 +16,39 @@ namespace nbl::video
 class CVulkanConnection final : public IAPIConnection
 {
 public:
-    CVulkanConnection(core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t appVer,
-        const char* appName, const SDebugCallback& dbgCb) : IAPIConnection(std::move(sys))
+    static core::smart_refctd_ptr<CVulkanConnection> create(
+        core::smart_refctd_ptr<system::ISystem>&& sys, uint32_t appVer, const char* appName,
+        bool enableValidation = true)
     {
-        VkResult result = volkInitialize();
-        assert(result == VK_SUCCESS);
-        
+        const std::vector<const char*> requiredInstanceLayerNames({ "VK_LAYER_KHRONOS_validation" });
+
+        // Todo(achal): Do I need to check availability for these?
+        // Currently all applications are forced to have support for windows and debug utils messenger
+        const uint32_t instanceExtensionCount = 3u;
+        const char* instanceExtensions[instanceExtensionCount] = { "VK_KHR_surface", "VK_KHR_win32_surface", VK_EXT_DEBUG_UTILS_EXTENSION_NAME };
+
+        if (volkInitialize() != VK_SUCCESS)
+        {
+            printf("Failed to initialize volk!\n");
+            return nullptr;
+        }
+
         VkApplicationInfo applicationInfo = { VK_STRUCTURE_TYPE_APPLICATION_INFO };
+        applicationInfo.pNext = nullptr; // pNext must be NULL
+        applicationInfo.pApplicationName = appName;
+        applicationInfo.applicationVersion = appVer;
+        applicationInfo.pEngineName = "Nabla";
         applicationInfo.apiVersion = VK_MAKE_VERSION(1, 1, 0);
         applicationInfo.engineVersion = NABLA_VERSION_INTEGER;
-        applicationInfo.pEngineName = "Nabla";
-        applicationInfo.applicationVersion = appVer;
-        applicationInfo.pApplicationName = appName;
-        
-        // Todo(achal): Get this from the user
-        const std::vector<const char*> requiredInstanceLayerNames({ "VK_LAYER_KHRONOS_validation" });
-        assert(areAllInstanceLayersAvailable(requiredInstanceLayerNames));
+
+        if (enableValidation)
+        {
+            if (!areAllInstanceLayersAvailable(requiredInstanceLayerNames))
+            {
+                printf("Validation layers requested but not available!\n");
+                return nullptr;
+            }
+        }
 
         // Note(achal): This exact create info is used for application wide debug messenger for now
         VkDebugUtilsMessengerCreateInfoEXT debugUtilsMessengerCreateInfo = { VK_STRUCTURE_TYPE_DEBUG_UTILS_MESSENGER_CREATE_INFO_EXT };
```

Comment:
OpenGL and Vulkan need some user-side control over how much messages and what messages the API should call the callback with @sadiuk @achalpandeyy